src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -77,10 +78,51 @@ class IterationCompletionCheck final {
         }
     }
 
-    explicit IterationCompletionCheck(PhaseContext& phaseContext)
-        : IterationCompletionCheck(phaseContext.get<TimeSpec, false>("Duration"),
-                                   phaseContext.get<UIntSpec, false>("Repeat"),
-                                   phaseContext.isNop()) {}
+    explicit IterationChecker(PhaseContext& phaseContext)
+        : IterationChecker(phaseContext.get<TimeSpec, false>("Duration"),
+                           phaseContext.get<IntegerSpec, false>("Repeat"),
+                           phaseContext.isNop()) {
+        auto rateSpec = phaseContext.get<RateSpec, false>("Rate");
+        const auto rateLimiterName =
+            phaseContext.get<std::string, false>("RateLimiterName").value_or("defaultRateLimiter");
+
+        if (rateSpec) {
+            _rng = std::make_optional(phaseContext.workload().createRNG());
```

Comment:
not sure you need the relatively heavyweight genny RNG to generate the jitter. I think you could safely use `rand() % 10000`